### DISCUSSION TITLE
How to get the description and the name value from the code API?

### DISCUSSION THREAD

**Discussion by biapar** [leekelleher/umbraco-contentment#364] [Question]
How to get the description and the name value from the code API?

**Comment by leekelleher** [comment]
Hi @biapar, do you mean for the Data List editor, as in when you have the selected value (from a content property)?

If I understand correctly, there isn't a way built-in to the Data List to reverse lookup the list items from a selected value. It's generally left to the implementor to figure that part out, through their own code.

If you are able to provide more information about your set-up/configuration, which data-source you are using, etc. then we might be able to offer some code examples of how to tackle it.

**Comment by biapar** [comment]
Hi, thanks for you reply. I mean that if I've a datalist with key as value and description as the text displayed into the dropdown, I wish the "description". I solved changing the query that the key is "key+description".

**Reply by leekelleher** [reply]
Is this for a custom data-source or one of the built-in data-source?

If it's a custom data-source, then where do you get the data from to popular the data-source? Could there be a way to develop a service or repository code that could take the selected value and return the associated name/description? (or other meta-data)

If it's for one of the built-in ones, let me know which one, as I might be able to help with an example code snippet.
